server/src/operators/chunk_operator.rs

```diff
@@ -1658,53 +1658,37 @@ pub fn get_highlights_with_exact_match(
                         .take(end_valid_boundary - start_valid_boundary)
                         .collect::<String>();
 
-                    let text_between_splits_word_count =
-                        text_between_splits.split_whitespace().count() as u32;
-                    let mut half_window_respecting_max_length = if (half_window * 2)
-                        + text_between_splits_word_count
-                        > max_length.unwrap_or(8)
-                    {
-                        let max_window = std::cmp::max(
-                            1,
-                            max_length.unwrap_or(8) as i32 - text_between_splits_word_count as i32,
-                        ) as u32;
-                        (max_window / 2) as usize
-                    } else {
-                        half_window as usize
-                    };
```

Comment:
Removing logic to restrict the window such that `highlight_max_length` is respected